client/src/components/edit-user-component/edit-user-component.jsx

```diff
@@ -0,0 +1,128 @@
+    // Create a JSON object with the required keys and values
+    const userData = {
+      firstName,
+      lastName,
+      email,
+      password,
+    };
+
+    // Convert the JSON object to a string
+    const userDataJSON = JSON.stringify(userData);
+
+    // Replace url with target route
+    fetch("http://localhost:8000/edit", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: userDataJSON,
+    });
+    // .then((response) => {
+    //     if (response.ok) {
+    //         // Handle success response (e.g., redirect or show a success message)
+    //         console.log('Login successful');
+    //     } else {
+    //         // Handle error response (e.g., show an error message)
+    //         console.error('Login failed');
+    //     }
+    // })
+    // .catch((error) => {
+    //     console.error('Error:', error);
+    // });
+  };
+
+  return (
+    <Form>
+      <Stack maxWidth="md" gap={4}>
+        <Header>
+          <Typography variant="h3">Edit User</Typography>
+          <Typography variant="body1">
+            Edit existing user information
+          </Typography>
+        </Header>
+
+        <Card>
+          <CardHeader title="User Information" />
+          <CardContent>
+            <Stack gap={1.5}>
+              <TextField
+                type="text"
+                id="firstName"
+                name="firstName"
+                value={firstName}
+                onChange={(e) => setFirstName(e.target.value)}
+                label="First Name"
+                helperText="*Required"
+                required
+              />
+              <TextField
+                type="text"
+                id="lastName"
+                name="lastName"
+                value={lastName}
+                onChange={(e) => setLastName(e.target.value)}
+                label="Last Name"
+                helperText="*Required"
+                required
+              />
+              <TextField
+                type="email"
+                id="email"
+                name="email"
+                value={email}
+                onChange={(e) => setEmail(e.target.value)}
+                label="Email"
+                helperText="*Required"
+                required
+              />
+              <TextField
+                type="password"
+                id="password"
+                name="password"
+                value={password}
+                onChange={(e) => setPassword(e.target.value)}
+                label="Password"
+                helperText="*Required"
+                required
+              />
+            </Stack>
+          </CardContent>
+        </Card>
+        <Stack direction="row">
+          <Cancel variant="outlined" size="large">
+            Cancel
+          </Cancel>
+          <Button
+            type="submit"
+            variant="contained"
+            size="large"
+            onClick={handleSave}
```

Comment:
Currently, despite the form validation onClick is called on the button. Instead of assigning the button's onClick method to handleClick, instead assign the  encompassing form's onSave method to it. This should be done in the create component as well.

Reply:
I couldn't find anything about an onSave method, but I added it to onSubmit.

Reply:
Whoops, thats what I meant to say!